.github/workflows/build_image.yml

```
@@ -0,0 +1,52 @@
+name: Build image
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "v*"
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+env:
+  DOCKER_BUILDKIT: 1
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build_image:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - amd64
+    steps:
+      - uses: actions/checkout@v3
+      - uses: docker/metadata-action@v4
+        id: meta
+        with:
+          images: increments/camo
+      - uses: docker/setup-qemu-action@v2
+      - uses: docker/setup-buildx-action@v2
+      - uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+      - name: Build and push
+        id: build
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          push: true
```

imo: I do not want to push to the registry every time I create a pull request.
I suggest that you only push under certain conditions.

```suggestion
          push: ${{ github.event_name != 'pull_request' }}
```

---

I want to push image to the registry for pull requests in order to test changes in our infra. 

docker/metadata-actions creates tags like `pr-2` for pull requests. (https://github.com/docker/metadata-action#basic)

---

I'm concerned about the creation of tags such as `pr-2` on [this page](https://hub.docker.com/r/increments/camo/tags).
However, I feel that untagging it after the validation is done will not cause any problems, so I agree with your it.

---

> However, I feel that untagging it after the validation is done will not cause any problems, so I agree with your it.

There are some actions to prune outdated `pr-*` tags introduced here :-) https://github.com/orgs/community/discussions/26267